packages/node/src/indexer/indexer.manager.ts

```diff
@@ -158,28 +177,33 @@ export class IndexerManager {
       });
     this.filteredDataSources = this.filterDataSources();
     this.fetchService.register((block) => this.indexBlock(block));
+
+    // Start VM for each data source
+    this.vms = await Promise.all(
```

Comment:
Can you shared IndexerSandbox across ds if their entries are same?

Reply:
Yeah, I wanted to do this too, but found it a little hard to identify a ds as going off the index doesn't work because we filter. 

I want to write some tests for this class then I will reassess sharing the VM